### ISSUE TITLE
Changing the Month string

### ISSUE THREAD

**Issue by neobie** [KidSysco/jquery-ui-month-picker#39]
Hi, thanks for the wonderful Month Picker.
I wish to know how can I remove the dot behind month ? eg. "Jan." to "Jan".
And further, how can I change it to other language?

**Comment by benjamin-albert** [comment]
You can use the built in i18n support.

Here is an example for translating the menu to hebrew:
```
$(".selector").MonthPicker({
        // Use the IsRTL option if you are translating the menu to a 
        // language the is written from right to left like hebrew or arabic.
        IsRTL: true,
        i18n: {
            year: 'שנת',
            buttonText: 'פתח תפריט',
            prevYear: "שנה קודמת",
            nextYear: "שנה הבאה",
            next12Years: 'עבור 12 שנים קדימה',
            prev12Years: 'עבור 12 שנים אחורה',
            nextLabel: "הבא",
            prevLabel: "הקודם",
            jumpYears: "בכר שנה",
            backTo: "חזור ל",
            months: ["ינו'", "פבר'", "מרץ", "אפר'", "מאי", "יוני", "יולי", "אוג'", "ספט'", "אוק'", "נוב'", "דצמ'"]
        }
});
```

**Comment by benjamin-albert** [comment]
@KidSysco I really think we should improve the documentation and add a demo for this.

**Comment by KidSysco** [comment]
ok I can take that one on if you want!

**Comment by benjamin-albert** [comment]
@KidSysco Yah sure go ahead, if you can also update the image in the README while your at it that would be great.

**Comment by KidSysco** [comment]
I am thinking of moving all docs to the wiki. I have started the process, take a look and let me know what you think Benjamin. I just think that the docs are getting too long, hard to manage and read as they are. That might be why people are not seeing things too. It might take me another day or two to get through the move. Let me know.

**Comment by neobie** [comment]
Thank you Benjamin, that is useful information.
And Thank you KidSysco for the plugin :100:

**Comment by benjamin-albert** [comment]
I absolutely agree that we should move the projects documentation out of the README.

But I'm not sure that GitHub wiki's will fit in to our workflow because there is [no pull request support](http://stackoverflow.com/questions/10642928/how-to-pull-request-a-wiki-page-on-github) (There's a workaround for this) and I don't feel comfortable editing the live documentation.

I personally think our goal should be to move the documentation and demo to [Github pages](https://pages.github.com).

Also in my experience [Github pages](https://pages.github.com) has significantly better [SEO](https://en.wikipedia.org/wiki/Search_engine_optimization).

@KidSysco Tell me what you think about Github pages.

**Comment by benjamin-albert** [comment]
In my experience people don't read, they scan (not just documentation but on the web in general).

For example if the user will search for 'selected month' in the wiki's first page they won't find anything, but if they search for it in the projects README they will most likely find what they are looking for.

I think we should organize the documentation the same way jQuery UI and Wikipedia does:
* A quick summery at the top.
* A table of contents (options, methods and events).
* All the details.

This way the user can just search for what they are looking for.

TLDR; [Myth #1: People read on the web](http://uxmyths.com/post/647473628/myth-people-read-on-the-web).

**Comment by KidSysco** [comment]
This is what we were originally going for. But I just think that
maintaining it is a real drag. All options should be in Alphabetical Order
and you can see that the Wiki is doing that for us. I agree that ideally,
the docs should be identical to the jQuery UI docs but it just seems like
they are maintaining documentation the hard way.

Finding the code to edit is hard with everything in one big page like that.

So my thoughts were that we would keep the format as identical as we can
but divide it up so that it is easier to maintain.

I am going to take a look at GitHub pages too and report back here.

Let me know what you think.

I will keep you posted.

r.

**Comment by KidSysco** [comment]
I am thinking we should use GitHub Pages as a way to advertise the plugin
and give it a nicely displayed front end.

We also should consider putting the demos there, so they just work.

Fiddle is still good and it is a great way to encourage users to submit
bugs, but I think GitHub pages will be better for a Demo Site.

Then for the docs, I am just thinking of using the GitHub wiki. We could
link from GitHub pages to the WIKI for documentation.

The Wiki breaks things up nicely, it is super easy to browse, and edit,
which is why I think it will work better. As I flip through it, I can see
changes that need to be made and fix them right there. I know there is no
source control on it but I am not sure it is really needed as it kinda gets
in the way.

You always ask me to read through your docs and see if the English looks
OK, and I do skim them but I don't really read it. It is too long of a doc
to read and I start skipping parts and missing things. Ever since I put it
into the WIKI I have noticed some things that need to be fixed so I went
ahead and made the changes.

Not only do I think it will help people find the docs they need but I think
it will help us keep the docs up to date if we use the WIKI. It is just too
easy and leave no excuses.

I have made you a Contributor Benjamin so you can make changes to the wiki
as you see fit.

I am going to continue to mess with it for a while, feel free to jump in
anytime, let me know your thoughts.

r.

**Comment by benjamin-albert** [comment]
Thanks for making me a Collaborator, I was thinking of asking for it so I can add tags/close issues and peek in the traffic graphs.

> I know there is no source control ...

Actually GitHub wikis are just git repositories that you can copy the git clone URL form under the pages outline and create branches locally if you want.

Also when edit a page there is a Page History button which show all off the commits (edits) made to that page.

I also really like that the documentation is broken up into pieces like this, I just wish that users has a way to search for text globally in them.

I will create most of the pages locally in a separate branch, this way documentation for features in development will not show up to users before they are finished.

What this means is that I will mention in future pull request to merge the feature branch with master in the wiki repository.

I've also noticed you've included i18n example in the internationalization, Localization with i18n and RTL support page.

@KidSysco Do you want me to work on moving the demo to GitHub pages while you move the rest of the README to the wiki or do you prefer work on the fiddle?

**Comment by benjamin-albert** [comment]
It looks like links that refer to a [Fragment Identifier](https://en.wikipedia.org/wiki/Fragment_identifier) (a.k.a hash) in order to refer the user to another page in the documentation don't work and need to be updated.

**Comment by KidSysco** [comment]
I appreciate your offer for help on this part too.

Since you seem to understand how we can get the wiki into source control,
then perhaps you should just get that rolling. Even if you don't finish it
all.

I can work on putting the fiddle code into the Demo Page, updating images,
etc.

I do have an image ready, here it is...

[image: Inline image 1]

**Comment by benjamin-albert** [comment]
@KidSysco The image is not showing, try uploading it from GitHub (and not from your email).

**Comment by KidSysco** [comment]
Heya Benjamin,

I have setup a GitHub page at the following URL.

https://kidsysco.github.io/jquery-ui-month-picker/

However, I have run into an issue. In IE only, if I hit the page, then scroll down a bit, then open up a month picker, the month picker shows up as far below the input as I scrolled. It only does this upon the first open of the menu. After that it works fine. I was wondering if you know what is going on there. It does NOT behave this way in jsFiddle.

I have been trying to spot the problem all weekend but I have been unable to find it. I was hoping to get a second set of eyes on it.

Here is the URL...

https://kidsysco.github.io/jquery-ui-month-picker/

There is another branch for this now too.

Let em know what you think.

r.

**Comment by KidSysco** [comment]
Even better yet. it does not behave this way in Edge. Only IE, only on MY web page.

Jeez I feel like I suck right now.

lol

**Comment by benjamin-albert** [comment]
Everything looks grate on Chrome, Firefox and Safari on Mac, I'll take a look at the problem in IE the next time I'm near a windows machine (when I get to work tomorrow).

I think it would be nice to add hidden code examples that the user can show by clicking View source (like jQuery UI does with their demos).

I cannot properly clone the repository on windows because the `.md` files for the options contain a `:` in them, which is not a void filename on windows.

This is a bit of a problem for me because I sometimes want to work on the documentation on my work machine, so I renamed `Options : MaxMonth.md` to `MaxMonth option.md`.

I've also created a [custom sidebar](https://github.com/KidSysco/jquery-ui-month-picker/wiki/_Sidebar/_edit) which allows us to group all option files under the `Options` category and control their order in the sidebar.

NOTE: When you create a new file in the wiki you will have to add it to the sidebar (which I really like because it allows hiding work in progress from users without having to clone the repository locally).

I also removed the duplicate `h3` title, there is not point in having MaxMonth option displayed as the title  (the filename) and than just MaxMonth again for no reason.

@KidSysco Tell me what you think about the new convention.

**Comment by KidSysco** [comment]
I agree, some code samples would be cool like they do on the jQuery UI website.

I will add those next!

I am going to close this issue out for now too. If there are any other ideas you have for the documentation or GitHub Page or anything at all, open another issue! Thanks for all of your help here again, we are really starting to make this a very nice repo with all of your latest ideas!
